tests/unit/pubsub_v1/subscriber/test_streaming_pull_manager.py

```diff
@@ -860,9 +956,72 @@ def test__on_response_modifies_ack_deadline():
         manager._on_response(response)
 
     dispatcher.modify_ack_deadline.assert_called_once_with(
-        [requests.ModAckRequest("ack_1", 18), requests.ModAckRequest("ack_2", 18)]
+        [requests.ModAckRequest("ack_1", 18, None), requests.ModAckRequest("ack_2", 18, None)]
+    )
+
+
+def test__on_response_modifies_ack_deadline_with_exactly_once_min_lease():
+    # exactly_once is disabled by default.
+    manager, _, dispatcher, leaser, _, scheduler = make_running_manager()
+    manager._callback = mock.sentinel.callback
+
+    # make p99 value smaller than exactly_once min lease
+    manager.ack_histogram.add(10)
+
+    # adjust message bookkeeping in leaser
+    fake_leaser_add(leaser, init_msg_count=0, assumed_msg_size=42)
+
+    # Set up the response with the first set of messages and exactly_once not
+    # enabled.
+    response1 = gapic_types.StreamingPullResponse(
+        received_messages=[
+            gapic_types.ReceivedMessage(
+                ack_id="ack_1",
+                message=gapic_types.PubsubMessage(data=b"foo", message_id="1"),
+            ),
+            gapic_types.ReceivedMessage(
+                ack_id="ack_2",
+                message=gapic_types.PubsubMessage(data=b"bar", message_id="2"),
+            ),
+        ],
+        subscription_properties = gapic_types.StreamingPullResponse.SubscriptionProperties(
+            exactly_once_delivery_enabled = False
+        )
+
+    )
+
+    # Set up the response with the second set of messages and exactly_once enabled.
+    response2 = gapic_types.StreamingPullResponse(
+        received_messages=[
+            gapic_types.ReceivedMessage(
+                ack_id="ack_3",
+                message=gapic_types.PubsubMessage(data=b"foo", message_id="1"),
+            ),
+            gapic_types.ReceivedMessage(
+                ack_id="ack_4",
+                message=gapic_types.PubsubMessage(data=b"bar", message_id="2"),
+            ),
+        ],
+        subscription_properties = gapic_types.StreamingPullResponse.SubscriptionProperties(
+            exactly_once_delivery_enabled = True
+        )
+
+    )
+
+    dispatcher.assert_has_calls(
+        # mod-acks called with histogram-based lease value
+        dispatcher.modify_ack_deadline([requests.ModAckRequest("ack_1", 10, None), requests.ModAckRequest("ack_2", 10, None)]),
+
+        # mod-acks called with 60 sec min lease value for exactly_once subscriptions
+        dispatcher.modify_ack_deadline([requests.ModAckRequest("ack_3", 60, None), requests.ModAckRequest("ack_4", 60, None)])
     )
 
+    # exactly_once is still disabled b/c subscription_properties says so
+    manager._on_response(response1)
+
+    # exactly_once should be enabled after this request b/c subscription_properties says so
+    manager._on_response(response2)
```

Comment:
It's not entirely clear what these two calls do, and there's no visible assertion. A comment explaining this would be useful.

Reply:
Sorry, I might be missing something, which comments were added? What is the role of `manager._on_response(...)` calls here, what errors/assertions do they trigger?